### PR TITLE
Support PerfCounters for Asp.Net Core apps if running inside Azure WebApps

### PR DESCRIPTION
**Pull request by cijothomas**
Fix Issue #889  .
SDK currently collects perf counters from Azure Web Apps using special ENV variables exposed by WebApps. This PR enables the same for .net core apps as well, which otherwise have no way of getting perf counters.

- [x] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Changes in public surface reviewed
- [ ] Design discussion issue #
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/CONTRIBUTING.md) instructions to build and test locally.

### CODE REVIEW

File: Src/PerformanceCollector/Perf.Shared/Implementation/WebAppPerformanceCollector/CacheHelper.cs [hunk 1]
```diff
@@ -1,7 +1,12 @@
 ﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.WebAppPerformanceCollector
 {
     using System;
+    using System.Collections.Generic;
+#if NETSTANDARD1_6
+    using Microsoft.Extensions.Caching.Memory;
+#else
```
Comment (cijothomas) [inline Src/PerformanceCollector/Perf.Shared/Implementation/WebAppPerformanceCollector/CacheHelper.cs]: Synced with sergey about this. We want to minimize adding additional dependencies for web sdk in .net framework target. So keeping existing behavior.
For netstandard, we take a new dependency of the new CacheProvider, and hence code is IF_DEF ed.

File: Src/PerformanceCollector/Perf.Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs [hunk 2]
```diff
@@ -153,7 +153,7 @@ private static double CollectCounter(string coutnerOriginalString, ICounterValue
                  throw new InvalidOperationException(
                     string.Format(
                         CultureInfo.CurrentCulture,
-                        Resources.WebAppPerformanceCounterReadFailed,
+                        "Failed to perform a read for web app performance counter {0}",
```
Comment (SergeyKanzhelev) [inline Src/PerformanceCollector/Perf.Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: Can you fix all the messages in a separate PR? Just to make everything consistent

Comment (cijothomas) [reply Src/PerformanceCollector/Perf.Shared/Implementation/WebAppPerformanceCollector/WebAppPerformanceCollector.cs]: Yes will do it.

File: Src/PerformanceCollector/NetCore/Stubs/Implementation/StandardPerformanceCollector/StandardPerformanceCollector.cs [hunk 3]
```diff
@@ -12,11 +12,19 @@ internal class StandardPerformanceCollector : IPerformanceCollector
...
+        {
+            PerformanceCollectorEventSource.Log.PerfCounterNetCoreOnlyOnAzureWebApp();
+        }
+
         /// <summary>
         /// Performs collection for all registered counters.
         /// </summary>
         /// <param name="onReadingFailure">Invoked when an individual counter fails to be read.</param>
-        public IEnumerable<Tuple<PerformanceCounterData, double>> Collect(Action<string, Exception> onReadingFailure = null)
+            public IEnumerable<Tuple<PerformanceCounterData, double>> Collect(Action<string, Exception> onReadingFailure = null)
```
Comment (tokaplan) [inline Src/PerformanceCollector/NetCore/Stubs/Implementation/StandardPerformanceCollector/StandardPerformanceCollector.cs]: Left-over tab?

Comment (cijothomas) [reply Src/PerformanceCollector/NetCore/Stubs/Implementation/StandardPerformanceCollector/StandardPerformanceCollector.cs]: addressed in a followup pr.

**Pull request merged**